bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+        FastOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd == OnOffType.ON) {
+                    int level = getMaxLightLevel(conf, 0xff);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) 0x12, (byte) level, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent fast on to switch {} level {}", nm(), dev.getAddress(),
+                            level == 0xff ? "on" : level);
+                } else if (cmd == OnOffType.OFF) {
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) 0x14, (byte) 0x00, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent fast off to switch {}", nm(), dev.getAddress());
+                }
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    @NonNullByDefault
+    public static class RampOnOffCommandHandler extends RampCommandHandler {
+        RampOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd == OnOffType.ON) {
+                    double ramptime = getRampTime(conf, 0);
+                    int ramplevel = getRampLevel(conf, 100);
+                    byte cmd2 = encode(ramptime, ramplevel);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, getOnCmd(), cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent ramp on to switch {} time {} level {} cmd1 {}", nm(), dev.getAddress(),
+                            ramptime, ramplevel, getOnCmd());
+                } else if (cmd == OnOffType.OFF) {
+                    double ramptime = getRampTime(conf, 0);
+                    int ramplevel = getRampLevel(conf, 0 /* ignored */);
+                    byte cmd2 = encode(ramptime, ramplevel);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, getOffCmd(), cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent ramp off to switch {} time {} cmd1 {}", nm(), dev.getAddress(), ramptime,
+                            getOffCmd());
+                }
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.